testsuite/cluster-test/src/experiments/recovery_time.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use std::{collections::HashSet, fmt, time::Duration};
+
+use anyhow::Result;
+use futures::future::{BoxFuture, FutureExt};
+use rand::Rng;
+use structopt::StructOpt;
+use tokio::time;
+
+use crate::cluster::Cluster;
+use crate::effects::{DeleteLibraData, Effect, StopContainer};
+use crate::experiments::Context;
+use crate::instance::Instance;
+use crate::tx_emitter::{EmitJobRequest, EmitThreadParams};
+use crate::{effects::Action, experiments::Experiment, stats};
+use slog_scope::info;
+use std::time::Instant;
+
+#[derive(StructOpt, Debug)]
+pub struct RecoveryTimeParams {
+    #[structopt(
+        long,
+        default_value = "100",
+        help = "Number of accounts to mint before starting the experiment"
+    )]
+    pub num_accounts_to_mint: u64,
+}
+
+pub struct RecoveryTime {
+    params: RecoveryTimeParams,
+    instance: Instance,
+}
+
+impl RecoveryTime {
+    pub fn new(params: RecoveryTimeParams, cluster: &Cluster) -> Self {
+        let rnd_index = rand::thread_rng().gen_range(0, cluster.instances().len());
+        let instance = cluster.instances()[rnd_index].clone();
+        Self { params, instance }
+    }
+}
+
+impl Experiment for RecoveryTime {
+    fn affected_validators(&self) -> HashSet<String> {
+        let mut result = HashSet::new();
+        result.insert(self.instance.short_hash().clone());
+        result
+    }
+
+    fn run(&mut self, context: &mut Context) -> BoxFuture<Result<Option<String>>> {
```

Comment:
You won't need all those `clone()` calls below if you annotate references with lifetimes:
```
    fn run<'a>(&'a mut self, context: &'a mut Context) -> BoxFuture<'a, Result<Option<String>>> {
```